src/components/v5/common/ActionSidebar/partials/TransactionTable/TransactionTable.tsx

```diff
@@ -1,73 +1,63 @@
 import React, { FC } from 'react';
-import { useIntl } from 'react-intl';
-import { v4 as uuidv4 } from 'uuid';
+import { FormattedMessage } from 'react-intl';
+import { useFieldArray, useFormContext, useWatch } from 'react-hook-form';
 
 import Button from '~v5/shared/Button/Button';
-import { useColonyContext } from '~hooks';
-import styles from './TransactionTable.module.css';
-import TransactionItem from '../TransactionItem';
-import { useTransactionTable } from './hooks';
+import { useColonyContext, useMobile } from '~hooks';
+import TableWithMeatballMenu from '~v5/common/TableWithMeatballMenu';
+import { useTransactionTableColumns, useGetTableMenuProps } from './hooks';
+import { TransactionTableModel, TransactionTableProps } from './types';
 
 const displayName = 'v5.common.ActionsContent.partials.TransactionTable';
 
-const TransactionTable: FC = () => {
-  const { formatMessage } = useIntl();
-  const { colony } = useColonyContext();
+const TransactionTable: FC<TransactionTableProps> = ({ name }) => {
+  const { nativeToken } = useColonyContext().colony || {};
+  const fieldArrayMethods = useFieldArray({
+    name,
+  });
+  const data: TransactionTableModel[] = fieldArrayMethods.fields.map(
+    ({ id }) => ({
+      key: id,
+    }),
+  );
+  const columns = useTransactionTableColumns(name);
+  const isMobile = useMobile();
+  const value = useWatch({ name });
+  const getMenuProps = useGetTableMenuProps(fieldArrayMethods, value);
   const {
-    transactions,
-    handleRemoveRowClick,
-    handleDuplicateRowClick,
-    updateTransaction,
-    setTransactions,
-  } = useTransactionTable();
-
-  const { nativeToken } = colony || {};
+    formState: { errors },
+  } = useFormContext();
 
   return (
     <div className="mt-7">
-      {!!transactions.length && (
-        <div className="mb-6">
-          <h5 className="text-2 mb-3">
-            {formatMessage({ id: 'actionSidebar.additionalPayments' })}
-          </h5>
-          <div className="border border-gray-200 rounded-lg">
-            <div className={styles.tableHead}>
-              <span className="w-1/3">Recipent</span>
-              <span className="w-2/3">Amount</span>
-            </div>
-            <div>
-              {transactions.map((transaction) => (
-                <TransactionItem
-                  {...transaction}
-                  key={transaction.key}
-                  id={transaction.key}
-                  recipent={transaction.recipent}
-                  onRemoveClick={handleRemoveRowClick}
-                  onDuplicateClick={handleDuplicateRowClick}
-                  onUpdate={updateTransaction}
-                />
-              ))}
-            </div>
-          </div>
-        </div>
+      {!!data.length && (
+        <TableWithMeatballMenu<TransactionTableModel>
+          className="mt-7"
+          isError={!!errors[name]}
```

Comment:
Instead of adding such prop, you should simply pass a `className` using `clsx` when there's an error.